rclcpp/src/rclcpp/executor.cpp

```diff
@@ -266,7 +266,7 @@ Executor::execute_subscription(
   if (ret == RCL_RET_OK) {
     message_info.from_intra_process = false;
     subscription->handle_message(message, message_info);
-  } else if (ret != RCL_RET_SUBSCRIPTION_TAKE_FAILED) {
+  } else if (ret == RCL_RET_SUBSCRIPTION_TAKE_FAILED) {
```

Comment:
Yes, in the case of the error message the return code is neither `RCL_RET_OK` nor `RCL_RET_SUBSCRIPTION_TAKE_FAILED` (aka `RCL_RET_SUBSCRIPTION_NOTHING_TAKEN`). Therefore something else which is a real problem hence the message.